presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeIdenticalWindows.java

```diff
@@ -97,7 +97,7 @@ public PlanNode visitWindow(
         {
             checkState(!node.getHashSymbol().isPresent(), "MergeIdenticalWindows should be run before HashGenerationOptimizer");
             checkState(node.getPrePartitionedInputs().isEmpty() && node.getPreSortedOrderPrefix() == 0, "MergeIdenticalWindows should be run before AddExchanges");
-            checkState(node.getFrames().values().size() == 1, "More than one frame per WindowNode is not yet supported");
+            checkState(node.getWindowFunctions().values().size() == 1, "More than one frame per WindowNode is not yet supported");
```

Comment:
This is not valid check anymore. What's more it's incorrect in that shape.
We DO support many functions while we don't support multiple frames. It should be something similar to what I've added here: https://github.com/prestodb/presto/pull/5785/files#diff-57adaed44436187133508287acd0b3dbR86
Check if among all frames there is at most one distinct.